ArduCopter/GCS_Mavlink.cpp

```diff
@@ -826,6 +827,31 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         return MAV_RESULT_FAILED;
 #endif
 
+#if MODE_GUIDED_ENABLED == ENABLED
+     case MAV_CMD_NAV_LOITER_TURNS:
+        if (copter.flightmode->in_guided_mode())
+        {
+            Location circle_center = Location((int32_t)(packet.param5*1e7), (int32_t)(packet.param6*1e7), (int32_t)(packet.param7*100), copter.current_loc.get_alt_frame());
+            // if latitude and longidude are not given, we default them to current_loc
+            if (circle_center.lat == 0 && circle_center.lng == 0) {
+                circle_center.lat = copter.current_loc.lat;
+                circle_center.lng = copter.current_loc.lng;
+            }
+
+            // use current alt if not specified
+            if (circle_center.alt == 0) {
+                int32_t curr_alt;
+                if(copter.current_loc.get_alt_cm(circle_center.get_alt_frame(), curr_alt))
+                    circle_center.set_alt_cm(curr_alt, circle_center.get_alt_frame());
+            }
+            uint8_t circle_radius_m = packet.param3;
```

Comment:
Let's keep this as float for as long as possible.
```suggestion
            const float circle_radius_m = packet.param3;
```

Reply:
Fixed it!